ENIGMAsystem/SHELL/Universal_System/bufferstruct.cpp

```diff
@@ -1485,43 +1516,93 @@ void game_load_buffer(buffer_t buffer) {
   }
   ptr += 1;
 
   ptr += enigma::backgrounds.deserialize_self(ptr);
 
-  type = enigma::deserialize<unsigned char>(ptr);
+  type = enigma::bytes_serialization::internal_deserialize<unsigned char>(ptr);
   if (type != 0xef) {
     DEBUG_MESSAGE("Invalid enigma::backgrounds footer", MESSAGE_TYPE::M_FATAL_ERROR);
   }
   ptr += 1;
 
-  auto room_index = enigma::deserialize<int>(ptr);
+  auto room_index = enigma::bytes_serialization::internal_deserialize<int>(ptr);
   enigma_user::room_goto(room_index);
   ptr += sizeof(int);
+  }
+  else {
+  std::vector<int> active_ids{};
+  std::transform(enigma::instance_list.begin(), enigma::instance_list.end(), std::back_inserter(active_ids),
+                 [](auto &value) { return value.first; });
+  for (int id : active_ids) {
+    instance_destroy(id);
+  }
+  
+  std::vector<int> inactive_ids{};
+  std::transform(enigma::instance_deactivated_list.begin(), enigma::instance_deactivated_list.end(), std::back_inserter(inactive_ids),
+                 [](auto &value) { return value.first; });
+  for (int id : inactive_ids) {
+    instance_destroy(id);
+  }
+
+  std::string json;
+  for (std::size_t i=1; i< binbuff->data.size();i++) {
+    json += static_cast<char>(binbuff->data[i]);
+  }
+
+  std::string active_list_part= enigma::JSON_serialization::json_find_value(json,"instance_list");
+  std::vector<std::string> active_list= enigma::JSON_serialization::json_split(active_list_part,',');
+  for (std::size_t i = 0; i < active_list.size(); i++) {
+    int obj_ind = enigma::JSON_serialization::enigma_deserialize<int>(enigma::JSON_serialization::json_find_value(active_list[i],"object_index"));
+    int obj_id = enigma::JSON_serialization::enigma_deserialize<int>(enigma::JSON_serialization::json_find_value(active_list[i],"id"));
+    auto obj = enigma::instance_create_id(0, 0, obj_ind, obj_id);
+    obj->json_deserialize_self(active_list[i]);
+    obj->activate();
+  }
+
+  std::string inactive_list_part= enigma::JSON_serialization::json_find_value(json,"instance_deactivated_list");
+  std::vector<std::string> inactive_list= enigma::JSON_serialization::json_split(inactive_list_part,',');
+  for(std::size_t i=0;i<inactive_list.size();i++) {
+    int obj_ind = enigma::JSON_serialization::enigma_deserialize<int>(enigma::JSON_serialization::json_find_value(inactive_list[i],"object_index"));
+    auto obj = enigma::instance_create_id(0, 0, obj_ind, -1);
+    obj->json_deserialize_self(inactive_list[i]);
+    obj->deactivate();
+  }
+
+  std::string backgrounds_list = enigma::JSON_serialization::json_find_value(json,"backgrounds");
+  if(backgrounds_list=="") {
+    DEBUG_MESSAGE("Invalid enigma::backgrounds header", MESSAGE_TYPE::M_FATAL_ERROR);
+  }
+
+  enigma::backgrounds.json_deserialize_self(backgrounds_list);
+
+  int room_index = enigma::JSON_serialization::enigma_deserialize<int>(enigma::JSON_serialization::json_find_value(json,"room_index"));  
+  enigma_user::room_goto(room_index);
+  }
```

Comment:
nit: this whole block should be indented one more level and the else moved after the closing brace on the line above it